src/main/integrations/child-process.ts

```diff
@@ -0,0 +1,113 @@
+import { addBreadcrumb, captureMessage, getCurrentHub } from '@sentry/core';
+import { NodeClient } from '@sentry/node';
+import { Integration, Severity } from '@sentry/types';
+
+import { ALL_REASONS, ExitReason, onChildProcessGone, onRendererProcessGone } from '../electron-normalize';
+import { ElectronMainOptions } from '../sdk';
+
+interface ChildProcessOptions<T> {
```

Comment:
Why make this generic? It's only be used for ExitReasons right?

Reply:
watch me try and attempt that across PRs 🤣

Reply:
lol

We can just do it in another PR, np. Step by step!